frontend/components/events/ShareModal.tsx

```diff
@@ -0,0 +1,147 @@
+    toggleModal();
+  };
+
+  const handleEmailShare = () => {
+    const subject = "Check out this event!";
+    const body = `I found this interesting event here: 
+${currentURL}`;
+    window.location.href = `mailto:?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;
+    toggleModal();
+  };
+
+  const copyURL = async () => {
+    try {
+      await navigator.clipboard.writeText(currentURL);
+      alert("URL copied to clipboard");
+    } catch (err) {
+      console.error("Failed to copy URL to clipboard:", err);
+    }
+  };
+
+  const highlightURL = (event: { preventDefault: () => void; stopPropagation: () => void }) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const urlElement = document.getElementById("url");
+    if (urlElement) {
+      const range = document.createRange();
+      range.selectNode(urlElement);
+      window.getSelection()?.removeAllRanges();
+      window.getSelection()?.addRange(range);
+    }
+  };
+
+  return (
+    <div className="relative">
+      <div className="flex items-center bg-organiser-light-gray rounded-xl w-1/6 cursor-pointer" onClick={toggleModal}>
+        <div className="rounded-md text-lg px-4 py-2">Share</div>
+        <Image src={share_arrow} alt="Share arrow" className="w-9 h-5 mx-2" />
+      </div>
+
+      <Transition appear show={modalOpen} as={Fragment}>
+        <Dialog as="div" className="relative z-50" onClose={toggleModal}>
+          <TransitionChild
+            as={Fragment}
+            enter="ease-out duration-300"
+            enterFrom="opacity-0"
+            enterTo="opacity-100"
+            leave="ease-in duration-200"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+          >
+            <div className="fixed inset-0 bg-black bg-opacity-50" />
+          </TransitionChild>
+
+          <div className="fixed inset-0 overflow-y-auto z-50">
+            <div className="flex min-h-full items-center justify-center p-4">
+              <TransitionChild
+                as={Fragment}
+                enter="ease-out duration-300"
+                enterFrom="opacity-0 scale-95"
+                enterTo="opacity-100 scale-100"
+                leave="ease-in duration-200"
+                leaveFrom="opacity-100 scale-100"
+                leaveTo="opacity-0 scale-95"
+              >
+               <div className="w-full max-w-2xl transform overflow-hidden rounded-2xl bg-white p-6 text-left align-middle shadow-xl transition-all relative">
+                  <div className="flex justify-between items-center">
+                    <h3 className="text-lg font-medium leading-6 text-gray-900">
+                      Share
+                    </h3>
+                    <button
+                      className="text-gray-500 hover:text-gray-700"
+                      onClick={toggleModal}
+                    >
+                      <svg className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
```

Comment:
could you pull out these svgs to separate files? they become a bit messy in the one big file.